Avail DA adapter Integration into rollkit

chandiniv1 opened the pull request:
This PR consist DA adapter for the [Avail](http://availproject.org/).

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced functionality for interacting with an availability layer for data storage.
	- Launched a gRPC service for the availability layer.
	- Added a mock server for handling block data-related HTTP requests.

- **Documentation**
	- Provided comprehensive setup and usage instructions for the Avail DA package.

- **Tests**
	- Integrated actual test execution commands in the continuous integration workflow.
	- Separated unit test execution from other test types in the Makefile.

- **Chores**
	- Updated the Makefile to run a complete suite of tests.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

nashqueue commented:
Thank you ! Can you make sure that the CI passes ?

chandiniv1 commented:
> Thank you ! Can you make sure that the CI passes ?

@nashqueue I have fixed the CI passes

Manav-Aggarwal commented on avail_test.go:
```diff
@@ -0,0 +1,10 @@
+)
+
+func TestAvailDA(t *testing.T) {
+	t.Skip()
```
Please add relevant unit tests to ensure code robustness

Manav-Aggarwal reviewed (changes requested):
Could you explain how this code has been tested, specifically the `Submit` and `Get` methods? Otherwise, looks great, just missing unit tests.

chandiniv1 commented:
> Could you please explain how this code has been tested, specifically the `Submit` and `Get` methods? Otherwise, looks great, just missing unit tests.
> 
> The CI currently does not test any core functionality except linting checks. See [CI](https://github.com/rollkit/avail-da/blob/f59576ddaa86f61884d905fe3125382aeaa2c588/.github/workflows/test.yml#L29)

@Manav-Aggarwal apart from writing unit tests, we haven't tested the application since the current version of rollkit is not compatible with GM application.

MSevey commented on go.mod:
```diff
@@ -1,3 +1,20 @@
 module github.com/rollkit/avail-da
 
-go 1.21.0
+go 1.21.1
```
We should add a README with this PR explaining how to use the interface, including the required go version.

MSevey reviewed (commented):
lgtm, minor comment about adding README and will re-review once unit tests are added.

chandiniv1 commented:
@Manav-Aggarwal @MSevey  I have added the readme regarding the usage of the da interface, Please have a look

chandiniv1 commented:
> > @Manav-Aggarwal @MSevey I have added the readme regarding the usage of the da interface, Please have a look
> 
> @chandiniv1 thanks for adding unit tests! Can you enable the tests by removing the echo command and uncommenting the code here:
> 
> https://github.com/rollkit/avail-da/blob/f59576ddaa86f61884d905fe3125382aeaa2c588/.github/workflows/test.yml#L29
> 
> Also I don't see the README, did it maybe not get committed?
> 
> Lastly, looks like golangci-lint is failing, can you address that please?

@MSevey I have enabled tests and fixed lint errors, please have a look

MSevey commented on .github/workflows/test.yml:
```diff
@@ -50,4 +49,4 @@ jobs:
           go-version: ${{ inputs.GO_VERSION }}
       - name: Integration Tests
-        run: echo "No integration tests yet"
+        run: make test
```
```suggestion
        run: echo "No integration tests yet"
```

MSevey reviewed (approved):
Great work!  Thanks!

tuxcanfly commented on avail.go:
```diff
@@ -0,0 +1,254 @@
+		if err != nil {
+			return nil, err
+		}
+		time.Sleep(10 * time.Second)
```
This sleeps in the first iteration as well, right? Might be better to move this to the `else ... == PROCESSING_BLOCK` condition.

tuxcanfly commented on go.mod:
```diff
@@ -1,3 +1,26 @@
+toolchain go1.21.5
+
+require (
+	github.com/rollkit/go-da v0.0.0-20231024133951-57bc36006772
```
go-da latest `v0.2.0` adds context and gas price for submit. Something to note for later.

tuxcanfly commented on go.mod:
```diff
@@ -1,3 +1,26 @@
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+)
+
+replace github.com/rollkit/go-da => github.com/rollkit/go-da v0.0.0-20231207150926-93600f28d67d
```
This seems unnecessary, just use the commit in the require clause.

tuxcanfly reviewed (dismissed):
Good work! Added some suggestions that can go in the next round.

Manav-Aggarwal reviewed (dismissed):
proxy utACK
